src/tests/GfxReplayTest.cpp

```diff
@@ -201,21 +224,35 @@ TEST(GfxReplayTest, player) {
 
   for (const auto keyframeIndex : keyframeIndicesToTest) {
     player.setKeyframeIndex(keyframeIndex);
+    int updatedNumberOfChildren = getNumberOfChildrenOfRoot(rootNode);
 
     if (keyframeIndex == -1) {
       // assert that lastRootChild doesn't have a sibling
       ASSERT(!lastRootChild->nextSibling());
     } else if (keyframeIndex == 0) {
       // assert that a new node was created under root
-      ASSERT(lastRootChild->nextSibling());
+      ASSERT(updatedNumberOfChildren > numberOfChildren);
     } else if (keyframeIndex == 1) {
       // assert that our stateUpdate was applied
-      ASSERT(lastRootChild->nextSibling());
-      const esp::scene::SceneNode* instanceNode =
-          static_cast<const esp::scene::SceneNode*>(
-              lastRootChild->nextSibling());
-      ASSERT(instanceNode->translation() == Mn::Vector3(1.f, 2.f, 3.f));
-      ASSERT(instanceNode->getSemanticId() == semanticId);
+      ASSERT(updatedNumberOfChildren > numberOfChildren);
+      if (numberOfChildren ==
+          0) {  // if rootNode had no children to begin with, then stateUpdate
+                // was applied to a new child node
+        const auto* rootChild = rootNode.children().first();
+        ASSERT(rootChild);
+        const esp::scene::SceneNode* instanceNode =
+            static_cast<const esp::scene::SceneNode*>(rootChild);
+        ASSERT(instanceNode->translation() == Mn::Vector3(1.f, 2.f, 3.f));
+        ASSERT(instanceNode->getSemanticId() == semanticId);
+      } else {  // if rootNode had children originally, then stateUpdate was
+                // applied to a sibling of lastRootChild
+        ASSERT(lastRootChild->nextSibling());
```

Comment:
I expected all these tests involving lastRootChild to be replaced with testing on the number of children.

Reply:
It seems like you're only using `lastRootChild` here in this else statement, for the case that `numberOfChildren` > 0. So maybe you can just make `lastRootChild` a local variable inside this else statement, instead of computing it earlier. I'd like us to avoid the `lastRootChild = &rootNode` because that just doesn't make sense to me (the root node is not a child).

Reply:
Thanks for the feedback @eundersander! I modified the comments of the test to be more clear, and use the number of children test wherever possible. I also got rid of `lastRootChild = &rootNode` and use the number of children to get the lastRootChild in the test with the new child. Hope this is more clear and let me know if there are any more changes you'd like to see!